ddtrace/tracer/log.go

```diff
@@ -0,0 +1,107 @@
+	SamplingRulesError    string            `json:"sampling_rules_error"`    // Any errors that occurred while parsing sampling rules
+	Tags                  map[string]string `json:"tags"`                    // Global tags
+	RuntimeMetricsEnabled bool              `json:"runtime_metrics_enabled"` // Whether or not runtime metrics are enabled
+	HealthMetricsEnabled  bool              `json:"health_metrics_enabled"`  // Whether or not health metrics are enabled
+	ApplicationVersion    string            `json:"dd_version"`              // Version of the user's application
+	Architecture          string            `json:"architecture"`            // Architecture of host machine
+	GlobalService         string            `json:"global_service"`          // Global service string. If not-nil should be same as Service. (#614)
+}
+
+// checkEndpoint tries to connect to endpoint, which should be the full URL
+// returned by transport.endpoint. If the endpoint is not reachable,
+// checkEndpoint returns an error explaining why.
+func checkEndpoint(endpoint string) error {
+	req, err := http.NewRequest("POST", endpoint, nil)
+	if err != nil {
+		return fmt.Errorf("cannot create http request: %v", err)
+	}
+	_, err = defaultClient.Do(req)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// logStartup generates a startupInfo for a tracer and writes it to the log in
+// JSON format.
+func logStartup(t *tracer) {
+	tags := make(map[string]string)
+	for k, v := range t.globalTags {
+		tags[k] = fmt.Sprintf("%v", v)
+	}
+
+	info := startupInfo{
+		Date:                  time.Now().Format(time.RFC3339),
+		OSName:                osName(),
+		OSVersion:             osVersion(),
+		Version:               version.Tag,
+		Lang:                  "Go",
+		LangVersion:           runtime.Version(),
+		Env:                   t.config.env,
+		Service:               t.config.serviceName,
+		AgentURL:              t.transport.endpoint(),
+		Debug:                 t.config.debug,
+		AnalyticsEnabled:      !math.IsNaN(globalconfig.AnalyticsRate()),
+		SampleRate:            fmt.Sprintf("%f", t.rulesSampling.globalRate),
+		SamplingRules:         t.rulesSampling.rules,
+		Tags:                  tags,
+		RuntimeMetricsEnabled: t.config.runtimeMetrics,
+		HealthMetricsEnabled:  t.config.runtimeMetrics,
+		ApplicationVersion:    t.config.version,
+		Architecture:          runtime.GOARCH,
+		GlobalService:         globalconfig.ServiceName(),
+	}
+	if _, err := samplingRulesFromEnv(); err != nil {
+		info.SamplingRulesError = fmt.Sprintf("%s", err)
+	}
+	if err := checkEndpoint(t.transport.endpoint()); err != nil {
+		info.AgentError = fmt.Sprintf("%s", err)
+		log.Warn("DIAGNOSTICS Unable to reach agent: %s", err)
+	}
+	bs, err := json.Marshal(info)
+	if err != nil {
+		log.Warn("DIAGNOSTICS Failed to serialize json for startup log (%v) %#v\n", err, info)
+		return
+	}
+	log.Info("CONFIGURATION %s\n", string(bs))
```

Comment:
If this needs to be the full string as specified by Luca, let's drop the whole `Info` stuff and just do `fmt.Println`

Reply:
I'm not sure if it does. I'm hoping we can get away with this to be more consistent with the rest of our tracer logs. If not, we'll drop the info and other prefix as you suggest.